src/main/kotlin/uk/gov/justice/digital/hmpps/supportadditionalneedsapi/domain/entity/ElspPlanEntity.kt

```diff
@@ -21,7 +21,7 @@ import java.util.*
 @EntityListeners(value = [AuditingEntityListener::class])
 @Table(name = "elsp_plan")
 @Audited(withModifiedFlag = false)
-data class ElspPlanEntity(
```

Comment:
data class automatically creates toString() which causes all stack overflow with plan -> plan contributors -> plan etc...

Reply:
yes (I think so) 

https://www.baeldung.com/kotlin/jpa#data-classes

This just says to be careful but there are other sources that say don't use them.